docs/application.rst

```diff
@@ -1,80 +1,75 @@
-Application Structure
-=====================
+The Warehouse codebase
+======================
 
-Note: this is a brain dump and its contents may be moved to a more
-appropriate location eventually.
+Warehouse uses the
+`Pyramid <https://docs.pylonsproject.org/projects/pyramid/en/latest/index.html>`__
+web framework, the
+`SQLAlchemy <https://docs.sqlalchemy.org/en/latest/>`__ ORM, and
+`Postgres <https://www.postgresql.org/docs/>`__ for its database.
+Warehouse's front end uses Jinja2 templates.
 
-At the moment it *just* lists the legacy structure and none of the intended
-new structure.
+The application exists within two `Docker <https://docs.docker.com/>`__
+containers, one of which contains static files for the website, and the
+other which contains the Python web application code running in a
+virtual environment and the database. In the development environment,
+`Docker Compose <https://docs.docker.com/compose/overview/>`__ manages
+running the containers and the connections between them.
```

Comment:
Technically this isn't true _yet_ -- we currently don't use any containers in production, and we use [far more than two containers in development](https://github.com/pypa/warehouse/blob/master/docker-compose.yml#L3), but this will be true some day in the near future.

It might be worth disambiguating the difference between how we use containers in a development environment vs. how we (will eventually) use them in production.